code/game/turfs/f13concrete.dm

```diff
@@ -0,0 +1,220 @@
+	description = "One of the needed ingrediants to make concrete and concrete shakes!"
+	color = "#A79E92"
+	taste_description = "cement"
+
+/datum/reagent/cement/on_mob_life(mob/living/carbon/M)
+//Don't eat cement, kids!
+	M.adjustOrganLoss(ORGAN_SLOT_STOMACH, 1*REAGENTS_EFFECT_MULTIPLIER, 150)
+	M.adjustToxLoss(0.2*REAGENTS_EFFECT_MULTIPLIER, 0)
+	M.adjustFireLoss(0.2*REAGENTS_EFFECT_MULTIPLIER, 0)
+	..()
+
+//////////////////
+//Concrete Shake//
+//////////////////
+/datum/chemical_reaction/concreteshake
+	name = "Concrete Shake"
+	id = "concrete_shake"
+	results = list(/datum/reagent/consumable/concreteshake = 5)
+	required_reagents = list(/datum/reagent/consumable/milk = 1, /datum/reagent/consumable/sugar = 1, /datum/reagent/cement = 1, /datum/reagent/consumable/ice = 1, /datum/reagent/consumable/cream = 1)
+
+/datum/reagent/consumable/concreteshake
+	name = "Concrete Shake"
+	description = "A mixture of icecream, milk and sugar along with... real cement powder?"
+	color = "#A79E92"
+	taste_description = "a bad idea"
+	glass_icon_state = "concrete_shake"
+	glass_name = "Concrete Shake"
+	glass_desc = "A thickshake served with a straw and a spoon. The spoon seems stuck and drinking through the straw takes serious effort."
+
+
+/datum/reagent/consumable/concreteshake/on_mob_life(mob/living/carbon/M)
+//Don't eat cement, kids!
+	M.adjustOrganLoss(ORGAN_SLOT_STOMACH, 1*REAGENTS_EFFECT_MULTIPLIER, 150)
+	M.adjustToxLoss(0.2*REAGENTS_EFFECT_MULTIPLIER, 0)
+	M.adjustFireLoss(0.2*REAGENTS_EFFECT_MULTIPLIER, 0)
+	..()
+
+
+////////////
+//Concrete//
+////////////
+/datum/reagent/concrete
+	name = "Concrete"
+	description = "Concrete"
+	color = "#A49483"
+
+/datum/chemical_reaction/concrete
+	name = "concrete"
+	id = "concrete"
+	required_reagents = list(/datum/reagent/cement = 1, /datum/reagent/water = 2, /datum/reagent/silicon = 7)
+
+/datum/chemical_reaction/concrete/on_reaction(datum/reagents/holder, multiplier)
+	var/location = get_turf(holder.my_atom)
+	for(var/i in 1 to multiplier)
+		new /obj/item/stack/sheet/mineral/concrete(location)
+//Makes one concrete sheet
+
+/datum/material/concrete
+	name = "concrete"
+	desc = "Concrete"
+	color = "#A49483"
+	sheet_type = /obj/item/stack/sheet/mineral/concrete
+
+/obj/item/stack/sheet/mineral/concrete
+	name = "concrete"
+	icon_state = "sheet-concrete"
+	singular_name = "concrete"
+	layer = LOW_ITEM_LAYER
+	novariants = FALSE
+	merge_type = /obj/item/stack/sheet/mineral/concrete
+	walltype = /turf/closed/wall/mineral/concrete
+
+/obj/item/stack/sheet/mineral/concrete/fifty
+	amount = 50
+
+/obj/item/stack/sheet/mineral/concrete/ten
+	amount = 10
+
+/obj/item/stack/sheet/mineral/concrete/five
+	amount = 5
+
+/obj/item/stack/sheet/mineral/concrete/get_main_recipes()
+	. = ..()
+	. += GLOB.concrete_recipes
+
+GLOBAL_LIST_INIT(concrete_recipes, list ( \
+	new/datum/stack_recipe("concrete barricade", /obj/structure/barricade/concrete, 2, time = 30, one_per_turf = 1, on_floor = 1), \
```

Comment:
Not sure what the 2 in this line does but I will assume its something intended

Reply:
The amount of concrete needed to craft the barricade!